spec/javascript/utils/compiler/variables.test.js

```diff
@@ -397,5 +394,68 @@ describe("variables.js", () => {
           .map((name) => [name, getDefaultVariableNameIndex(name)]))
       ).toStrictEqual(expectedOutput)
     })
+
+    test("Should get Global variables from parameter objects", () => {
+      completionsMap.set(setCompletionsMap("For Global Variable"))
```

Comment:
You can do this in a `beforeAll()` block, and it will [set the `completionsMap` before all the `test()`s run](https://jestjs.io/docs/setup-teardown#one-time-setup).

This means you can set all the Workshop Actions you want in the completionsMap at the beginning and not have to worry about making them available in each individual `test()`.

And don't worry, as long as you put the `beforeAll()` block inside the first `describe()` block in the file, your `completionsMap` should persist throughout all the tests (see [Scoping](https://jestjs.io/docs/setup-teardown#scoping) in our test runner docs).

Reply:
There is `beforeAll()`, which runs before all `test()`s under the same scope, and `beforeEach()`, which runs before each `test()` under the same scope.

I think its better because you can keep things nice and tidy. One part is setup (`beforeAll()`), and the other part is testing and verifying (all the `test()`s).
You can also say "the library provides it and is a common pattern, so it must be good"™️.